dracut-functions.sh

```diff
@@ -227,15 +227,23 @@ get_fs_env() {
     return 1
 }
 
+# Path to cache file, with failsafe in case the file is not sourced by dracut
+[ -n "$get_maj_min_cache_file" ] || get_maj_min_cache_file="$(mktemp)"
```

Comment:
Actually this test is just here to not have failure in case someone sources dracut-functions.sh directly, which then would make get_maj_min produce errors.

I see a lot of places where external variables are used in that script, so maybe this can just be removed.

Reply:
Then I would suggest to replace any `$get_maj_min_cache_file` with `${get_maj_min_cache_file:?}`

Reply:
OK, pushing new code